### ISSUE TITLE
Java interop tests are broken on master

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#8351]
It's been hapenning consistently for last X builds on master.

https://grpc-testing.appspot.com/job/gRPC_interop_master/8075/#showFailuresLink

```
stdin: is not a tty
Gradle is no longer run automatically. Make sure to run
'./gradlew installDist -PskipCodegen=true' or
'./gradlew :grpc-interop-testing:installDist -PskipCodegen' after any changes.
-PskipCodegen=true is optional, but requires less setup.
Oct 12, 2016 9:25:56 AM io.grpc.internal.ManagedChannelImpl <init>
INFO: [ManagedChannelImpl@70a9f84e] Created with target directaddress:///foo.test.google.fr/127.0.0.1:58073
Running test cancel_after_first_response
Exception in thread "main" java.lang.AssertionError: No record found
	at io.grpc.testing.integration.AbstractInteropTest.assertServerMetrics(AbstractInteropTest.java:1176)
	at io.grpc.testing.integration.AbstractInteropTest.assertMetrics(AbstractInteropTest.java:1120)
	at io.grpc.testing.integration.AbstractInteropTest.assertMetrics(AbstractInteropTest.java:1128)
	at io.grpc.testing.integration.AbstractInteropTest.cancelAfterFirstResponse(AbstractInteropTest.java:429)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:276)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:199)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:84)
Shutting down

FAILED: cloud_to_cloud:java:ruby_server:cancel_after_first_response [ret=1, pid=27773]
```

**Comment by jtattermusch** [comment]
CC @carl-mastrangelo

**Comment by ejona86** [comment]
I've got a fix. I'm sending out a PR.

**Comment by ejona86** [comment]
This is fixed by grpc/grpc-java#2330
